services/server/src/lib/availabilitySlot.ts

```diff
@@ -51,6 +56,58 @@ export async function deleteSlots({
   ]);
 }
 
+export async function updateSlot({
+  currentUserId,
+  updateAction,
+  tx,
+}: {
+  currentUserId: number;
+  updateAction: IAvailabilitySlot.UpdateAction;
+  tx: Knex.Transaction;
+}): Promise<void> {
+  const now = dayjs.utc();
+  const slot = await availabilitySlots.findById(updateAction.id);
```

Comment:
For slot you want to update, you will send a specific query to the database to get it. So updating 10 slots, will require 10 queries. We can get all the slots from the database in one query and then you can pass the slot you are intested in to the `updateSlot`.